src/Testcontainers.Papercut/PapercutContainer.cs

```diff
@@ -0,0 +1,76 @@
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Testcontainers.Papercut;
+
+/// <inheritdoc cref="DockerContainer" />
+[PublicAPI]
+public sealed class PapercutContainer : DockerContainer
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PapercutContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    /// <param name="logger">The logger.</param>
+    public PapercutContainer(PapercutConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+    }
+
+    public string WebUrl => "http://" + Hostname + ":" + GetMappedPublicPort(37408);
+
+    public async Task<PapercutMessageCollection> GetMessages()
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        var messageText = await client.GetStringAsync("/api/messages");
+        return JsonConvert.DeserializeObject<PapercutMessageCollection>(messageText);
+    }
+    
+    public async Task<PapercutMessage> GetMessage(string id)
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        var messageText = await client.GetStringAsync($"/api/messages/{id}");
+        return JsonConvert.DeserializeObject<PapercutMessage>(messageText);
+    }
+
+    public async Task<bool> DeleteMessages()
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        var response = await client.DeleteAsync("/api/messages");
+        return response.IsSuccessStatusCode;
+    }
+
+    public async Task<bool> DeleteMessages(string id)
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        var response = await client.DeleteAsync($"/api/messages/{id}");
+        return response.IsSuccessStatusCode;
+    }
+
+    public Task<HttpResponseMessage> DownloadRaw(string id)
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        return client.GetAsync($"/api/messages/{id}/raw");
+    }
+
+    public Task<HttpResponseMessage> DownloadSection(string id, int index)
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        return client.GetAsync($"/api/messages/{id}/sections/{index}");
+    }
+
+    public Task<HttpResponseMessage> DownloadSectionContent(string id, string contentId)
+    {
+        var client = new HttpClient();
+        client.BaseAddress = new Uri(WebUrl);
+        return client.GetAsync($"/api/messages/{id}/contents/{contentId}");
+    }
```

Comment:
Is there no Papercut client or, at the very least, a dependency that holds the domain models? Is there a publicly available API document we can refer to? The Testcontainers modules do not implement client functionalities; instead, we focus on spinning up dependencies for various use cases. Can we remove these client functionalities? We can move and keep the necessary domain models to the test and simply send the required HTTP requests from there.

Reply:
Cool, yeah, that makes sense responsibility wise all the container package is doing is creating the container, and interacting with it is outside the scope of this package. I have removed the HTTP Client calls from the package code (moved some/most of it to the tests)